### ISSUE TITLE
[Crash]: Crash with Potion Effect on Living Base Galacticraft Player Entity

### ISSUE THREAD

**Issue by Lanse505** [micdoodle8/Galacticraft#1365]
This crash has happened with Blood Magic, Witchery and RandomThings potion effects, Getting the effects has a completely random chance of continously crashing you. This crash is only happening if Galacticraft is installed and always points to the Galacticraft Player Entity.

Really hope this can get fixed.

Crashlog: 
http://pastebin.com/kp16MY4T

**Comment by radfast** [comment]
> This crash is only happening if Galacticraft is installed

Not true.  Go test it more carefully without Galacticraft installed and you will see.

* This type of crash is __NOT__ caused by Galacticraft.
*  Whenever any mod crashes the player, ```micdoodle8.mods.galacticraft.core.entities.player.GCEntityClientPlayerMP``` will always appear in your stacktrace if Galacticraft is installed, the same way that ```net.minecraft.client.entity.EntityClientPlayerMP``` is shown

The cause of your problem here is some other mod crashing with the potion effect.

**Comment by KirinDave** [comment]
Initially Galacticraft definitely seems implicated. The first test we ran was seeing the issue is reproducible with Galacticraft. We then removed Galacticraft from the resolution and the crash stopped happening.

It may be the case that someone else is not implementing potion effects properly and using something they're not supposed to, but it is also true that the issue goes away when Galacticraft isn't there.

If this isn't Galacticraft's issue, could we trouble you to suggest maybe how we might help track down who's it is? Because we can reproduce the issue with multiple mod potion effects, so we're not sure why it WOULDN'T be GC.

**Comment by radfast** [comment]
You're totally sure about this?  I find it incredibly surprising:
>  it is also true that the issue goes away when Galacticraft isn't there.

In that case tell me a test so I can reproduce the issue here, ideally with only Galacticraft + 1 other mod?

The cause of the crash appears to be a potion effect which is improperly registered, so this line in vanilla Minecraft code is returning null ```Potion.potionTypes[this.potionID]```  Galacticraft does not register any potion effects.

**Comment by radfast** [comment]
See also #1323 which looks the same.
Do you have either Infernal Mobs or Biomes o' Plenty installed?

**Comment by radfast** [comment]
and surprise surprise, looking at your crash report you do indeed have both Infernal Mobs and Biomes o' Plenty.  As covered on #1323 and #1366, the issue appears to be caused by the Infernal version of a BoP mob.  Not Galacticraft.

**Comment by Lanse505** [comment]
Removed Infernal Mobs, still crash ^^''

**Comment by radfast** [comment]
Something else then.  Which mob was attacking you when it crashed?

**Comment by Lanse505** [comment]
This has a 99% chance to happen with BoP Phantoms which might point to BoP Phantoms implementing some kind of code inproperly?, other times it has mostly only happened with Vanilla witches but thats like once in a blue moon

**Comment by radfast** [comment]
That's helpful info, so I can reproduce this by:
1. Spawning in a BoP Phantom
2. Standing there until it hits me with a poison effect?

**Comment by Lanse505** [comment]
Hmm, so BoP has some kind of issue with potion effects >.>, i'll go poke Adubbz and see if he know about it and will atleast fix it in 1.8.

**Comment by gixmy** [comment]
FWIW I'm getting the same crash - see http://www.minecraftforum.net/forums/mapping-and-modding/minecraft-mods/2076348-galacticraft-add-on-more-planets-beta-1-1-9?page=10#c189

It happens just after minecraft has loaded a More Planets dimension and I'm still floating down on my parachute. The other More Planets dimensions that I've tried work ok as do the Galacticraft ones.

I do not have BOP or Infernal Mobs installed.

**Comment by gixmy** [comment]
I've resolved the issue I was having. Googling around I found this crash happens with other mods too. Someone mentioned it could be related to Blood Magic. Removing the Blood Magic mod I no longer get a crash. Interestingly More Planets applies an "infected gas" effect (potion) when you arrive on the dimension where I was getting the crash. I think it was this conflicting with BloodMagic that caused the crash.
